### PR TITLE
MTDSA-5133 added request parser and validation for retrieve liabilities

### PR DESCRIPTION
**Pull request by david-walters-HMRC**

### CODE REVIEW

File: test/v1/controllers/requestParsers/LiabilityRequestParserSpec.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          Left(ErrorWrapper(None, InvalidToError, None))
+      }
+
+      "multiple request parameters supplied are invalid" in new Test {
+        MockVrnValidator.validate(LiabilityRawData(invalidVrn, invalidFrom, invalidTo))
+          .returns(List(VrnFormatError))
+
+        parser.parseRequest(LiabilityRawData(invalidVrn, invalidFrom, invalidTo)) shouldBe
+          Left(ErrorWrapper(None, VrnFormatError, None))
+      }
```
Comment (MarkAKelly) [inline test/v1/controllers/requestParsers/LiabilityRequestParserSpec.scala:80]: Add test to check the correct order when more than one invalid parameter is supplied

File: app/v1/controllers/requestParsers/validators/LiabilitiesValidator.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+package v1.controllers.requestParsers.validators
+
+import v1.controllers.requestParsers.validators.validations.{DateFormatValidation, DateRangeValidation, VrnValidation}
+import v1.models.errors.{InvalidFromError, InvalidToError, MtdError}
+import v1.models.request.liability.LiabilityRawData
+
+class LiabilitiesValidator extends Validator[LiabilityRawData]  {
+
+  private val validationSet = List(vrnFormatValidation, dateFormatValidation, dateRangeValidation)
```
Comment (MarkAKelly) [inline app/v1/controllers/requestParsers/validators/LiabilitiesValidator.scala:25]: check the order of this

File: test/v1/controllers/requestParsers/validators/RetrieveLiabilitiesValidatorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    "return only FromDateFormatError error" when {
+      "an invalid from date format is supplied" in {
+        validator.validate(LiabilityRawData(validVrn, "12-31-2020", validTo)) shouldBe List(InvalidFromError)
+      }
+    }
+
+    "return only ToDateFormatError error" when {
+      "an invalid to date format is supplied" in {
+        validator.validate(LiabilityRawData(validVrn, validFrom, "12-31-2020")) shouldBe List(InvalidToError)
+      }
```
Comment (MarkAKelly) [inline test/v1/controllers/requestParsers/validators/RetrieveLiabilitiesValidatorSpec.scala:60]: actually check the order here

File: app/v1/controllers/requestParsers/validators/validations/PaymentsLiabilitiesDateRangeValidation.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    val fmtTo = LocalDate.parse(to, dateFormat)
+    List(
+      checkIfDateRangeIsIncorrect(fmtFrom, fmtTo)
+    ).flatten
+  }
+
+  private def checkIfDateRangeIsIncorrect(from: LocalDate, to: LocalDate): List[MtdError] = {
+
+    if(!from.isBefore(to) || from.plusYears(1).minusDays(1).isBefore(to)) {
+      Logger.error(s"from.plusYears(1).minusDays(1) :\n" +
```
Comment (yangwong9664) [inline app/v1/controllers/requestParsers/validators/validations/PaymentsLiabilitiesDateRangeValidation.scala:37]: Can you check this log format

File: app/v1/controllers/requestParsers/validators/validations/PaymentsLiabilitiesDateRangeValidation.scala [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    val fmtFrom = LocalDate.parse(from, dateFormat)
+    val fmtTo = LocalDate.parse(to, dateFormat)
+    List(
+      checkIfDateRangeIsIncorrect(fmtFrom, fmtTo)
+    ).flatten
+  }
+
+  private def checkIfDateRangeIsIncorrect(from: LocalDate, to: LocalDate): List[MtdError] = {
+
+    if(!from.isBefore(to) || from.plusYears(1).minusDays(1).isBefore(to)) {
```
Comment (yangwong9664) [inline app/v1/controllers/requestParsers/validators/validations/PaymentsLiabilitiesDateRangeValidation.scala:36]: the error log below doesn't make sense for "!from.isBefore(to)". Maybe there should be 1 generic error log, or 2 separate logs

Comment (narendravijayarao) [reply app/v1/controllers/requestParsers/validators/validations/PaymentsLiabilitiesDateRangeValidation.scala:36]: Removed the log, we are not adding logs anywhere, why do we need here?

**Pull request merged**